simpleviewinc/keg-hub
# Keg-Components - Theme as Functions

[Semver-Status] major

> This PR should be reviewed along with [this PR](https://github.com/simpleviewinc/tap-events-force/pull/92)

##  Context

* Currently there is no way to override the theme defaults of keg-components
* Any dynamically generated values or states, must be replicated manually


## Goal

* Allow overwriting the theme defaults by converted all theme objects to functions
* The defaults are then pulled in when the function is called instead of on initial load
* This allows overwriting the default theme config before build the theme object

## Updates

* `repos/keg-components/.storybook/preview.js`
  * Added call to initialize theme
* `repos/keg-components/src/theme/*`
  * Converted all theme files into functions
  * Updated on theme initialization to call each theme function
  * Allows passing in a config object
  * Allow overwriting the theme defaults 
* `repos/keg-components/src/*/*.stories.js`
  * Updated the hierarchy
  * Cleaned up layout in storybook
  * Added sort helper
 

## Testing

* Run the docker package => `keg dpg run docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:keg-theme-as-functions`
* Navigate to this URL => http://components-keg-theme-as-functions.local.kegdev.xyz
* Ensure all components work as expected
  * There should be no difference between the components with these changes, and how they were before
